pkg/operator/certrotation/cabundle.go

```diff
@@ -41,18 +45,29 @@ type CABundleConfigMap struct {
 func (c CABundleConfigMap) ensureConfigMapCABundle(ctx context.Context, signingCertKeyPair *crypto.CA) ([]*x509.Certificate, error) {
 	// by this point we have current signing cert/key pair.  We now need to make sure that the ca-bundle configmap has this cert and
 	// doesn't have any expired certs
+	secretExists := true
 	originalCABundleConfigMap, err := c.Lister.ConfigMaps(c.Namespace).Get(c.Name)
 	if err != nil && !apierrors.IsNotFound(err) {
 		return nil, err
 	}
 	caBundleConfigMap := originalCABundleConfigMap.DeepCopy()
 	if apierrors.IsNotFound(err) {
 		// create an empty one
-		caBundleConfigMap = &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: c.Namespace, Name: c.Name}}
+		caBundleConfigMap = &corev1.ConfigMap{ObjectMeta: NewObjectMeta(
+			c.Namespace,
+			c.Name,
+			c.JiraComponent,
+			c.Description,
+		)}
+		secretExists = false
 	}
 	if c.Owner != nil {
 		ensureOwnerReference(&caBundleConfigMap.ObjectMeta, c.Owner)
 	}
+	if err := c.ensureTLSMetadata(ctx, caBundleConfigMap, secretExists); err != nil {
+		return nil, err
+	}
+
 	updatedCerts, err := manageCABundleConfigMap(caBundleConfigMap, signingCertKeyPair.Config.Certs[0])
```

Comment:
```suggestion
	needsMetadataUpdate := false
	if c.Owner != nil {
		ensureOwnerReference(&caBundleConfigMap.ObjectMeta, c.Owner)
	}
	if len(c.JiraComponent) > 0 || len(c.Description) > 0{
	    needsMetadataupdate = ensureTLSMetadata(&caBundleConfigMap.ObjectMeta, c.JiraComponent, c.Descrption) || needsMetadataupdate
	}
	if needsMetadataUpdate && secretExists{
	        _, _, err := resourceapply.ApplyConfigMap(ctx, c.Client, c.EventRecorder, configMap)
		if err != nil {
			return err
		}
	}

	updatedCerts, err := manageCABundleConfigMap(caBundleConfigMap, signingCertKeyPair.Config.Certs[0])
```

notice how it could (should?) be congruent to the ownerreference metadata change already possible.